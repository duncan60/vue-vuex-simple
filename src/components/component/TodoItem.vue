<template>
  <li v-if="!todo.isEdit && !todo.isComplete">
    <span>{{todo.text}}</span>
    <button @click="onHandler('complete')">
      complete
    </button>
    <button @click="onHandler('edit')">
      edit
    </button>
  </li>
  <li v-else-if="todo.isComplete">
    <span>{{todo.text}}</span>
    <button @click="onHandler('restore')">
      restore
    </button>
  </li>
  <li v-else>
    <input type="text" v-model="text">
    <button @click="onHandler('save')" :disabled="text === ''">
      save
    </button>
    <button @click="onHandler('cancel')">
      cancel
    </button>
  </li>
</template>

<script>
export default {
  name: 'todo-item',
  props: ['todo', 'updateTodo'],
  data() {
    return {
      text: this.todo.text,
    };
  },
  methods: {
    onHandler(type) {
      this.updateTodo(type, this.todo, this.text);
      this.text = this.todo.text;
    },
  },
};
</script>
