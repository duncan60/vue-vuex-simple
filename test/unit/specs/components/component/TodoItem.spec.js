import { mount, shallow } from '@vue/test-utils';
import TodoItem from '@/components/component/TodoItem';

let initProps = {};

beforeAll(() => {
  initProps = {
    todo: {
      text: 'testing',
      isEdit: false,
      isComplete: false,
    },
  };
});

describe('TodoItem.vue', () => {
  it('test initial rendering', () => {
    const wrapper = mount(TodoItem, {
      propsData: {
        ...initProps,
      },
    });
    const template = wrapper.html();
    expect(template).toMatchSnapshot();
  });

  it('test isEdit is true rendering', () => {
    const wrapper = mount(TodoItem, {
      propsData: {
        todo: {
          ...initProps.todo,
          isEdit: true,
        },
      },
    });
    const template = wrapper.html();
    expect(template).toMatchSnapshot();
  });

  it('test isComplete is true rendering', () => {
    const wrapper = mount(TodoItem, {
      propsData: {
        todo: {
          ...initProps.todo,
          isComplete: true,
        },
      },
    });
    const template = wrapper.html();
    expect(template).toMatchSnapshot();
  });

  it('test render text', () => {
    const component = shallow(TodoItem, {
      propsData: {
        ...initProps,
      },
    });
    const text = component.vm.$el.querySelector('span').textContent;
    expect(text).toEqual('testing');
  });
});
