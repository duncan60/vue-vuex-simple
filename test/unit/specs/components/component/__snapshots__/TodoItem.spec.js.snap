// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem.vue test initial rendering 1`] = `
<li><span>testing</span>
    <button>
        complete
    </button>
    <button>
        edit
    </button>
</li>
`;

exports[`TodoItem.vue test isComplete rendering 1`] = `
<li><span>testing</span>
    <button>
        restore
    </button>
</li>
`;

exports[`TodoItem.vue test isEdit rendering 1`] = `
<li>
    <input type="text">
    <button>
        save
    </button>
    <button>
        cancel
    </button>
</li>
`;
